code/game/machinery/washing_machine.dm

```diff
@@ -238,102 +307,160 @@
 			else
-				to_chat(user, "<span class='notice'>You can't put the item in right now.</span>")
+				to_chat(user, "<span class='notice'>The washing machine is is occupied by [locate(/mob,contents)], there's no extra space.</span>")
 		else
 			to_chat(user, "<span class='notice'>The washing machine is full.</span>")
 		update_icon()
 	else
 		return ..()
 
 /obj/machinery/washing_machine/attack_hand(mob/user as mob)
-	switch(state)
-		if(1)
-			state = 2
-		if(2)
-			state = 1
-			for(var/atom/movable/O in contents)
-				O.loc = src.loc
-		if(3)
-			state = 4
-		if(4)
-			state = 3
-			for(var/atom/movable/O in contents)
-				O.loc = src.loc
-			crayon = null
-			state = 1
-		if(5)
-			to_chat(user, "<span class='warning'>[src] is busy.</span>")
-		if(6)
-			state = 7
-		if(7)
-			if(gibs_ready)
-				gibs_ready = 0
+	if(running) // States 5 and 8
+		to_chat(usr, "<span class='warning'>\The [src] is busy.</span>")
+	else
+		if(bloody)
+			if(open) // 6
+				to_chat(usr, "<span class='notice'>\The [src] only contains blood, there is nothing to remove from it.</span>")
+			else // 7
 				if(locate(/mob,contents))
 					var/mob/M = locate(/mob,contents)
-					M.gib()
-			for(var/atom/movable/O in contents)
-				O.loc = src.loc
-			crayon = null
-			state = 1
-
+					if(!ishuman(M) && M.stat == DEAD)
+						M.gib()
+						to_chat(usr, "<span class='narsie'>You monster.</span>")
+				for(var/atom/movable/O in contents) // it empties here because gibs would explode when opening the door
+					O.loc = src.loc
+				crayon = null
+				full = FALSE
+				open = TRUE
+				playsound(src, 'sound/machines/click.ogg', 25)
+		else
+			if(full)
+				if(open) // 3
+					to_chat(usr, "<span class='notice'>You empty \the [src].</span>")
+					for(var/atom/movable/O in contents)
+						O.loc = src.loc
+					crayon = null
+					full = FALSE
+				else // 4
+					to_chat(usr, "<span class='notice'>You open the door on \the [src].</span>")
+					open = TRUE
+					playsound(src, 'sound/machines/click.ogg', 25)
+			else
+				if(open) // 1
+					to_chat(usr, "<span class='notice'>\The [src] is empty, there is nothing to remove from it.</span>")
+					for(var/atom/movable/O in contents) // We'll try anyways, just in case
+						O.loc = src.loc
+					crayon = null
+				else // 2
+					to_chat(usr, "<span class='notice'>You open the door on \the [src].</span>")
+					open = TRUE
+					playsound(src, 'sound/machines/click.ogg', 25)
 
 	update_icon()
 
+/obj/machinery/washing_machine/examine(mob/user)
+	. = ..()
+	var/contentsjoined = contents.Join(", ")
+	if(running)
+		if(bloody)
+			if(length(contents) == 0)
+				. += "<span class='warning'>\The [src] filled with blood!</span>"
+			else if(length(contents) > 1)
+				. += "<span class='warning'>\The [src] has [length(contents)] objects inside and tons of blood!</span>"
+			else
+				. += "<span class='warning'>\The [src] has 1 object inside and tons of blood!</span>"
```

Comment:
Use the \s macro to avoid having to make these extra conditionals
```suggestion
			else
				. += "<span class='warning'>\The [src] has [length(contents)] object\s inside and tons of blood!</span>"
```